File: Python_Scripting_for_Computational_Science_Third_Edition/perl/swap1.pl
#!/usr/bin/perl
$arg = "[^,]+";
$call = "superLibFunc\\s*\\(\\s*($arg)\\s*,\\s*($arg)\\s*\\)";
print $call;

@cfiles = ("../py/regex/.test1.c");
foreach $cfile (@cfiles) {
    print "Treating $cfile...\n";
    open(FILE,"<$cfile") || die "cannot open file: $!\n";
    @lines = <FILE>;  # load all lines into a list
    close(FILE);
    $filestr = join("",@lines);
    $filestr =~ s/$call/superLibFunc($2, $1)/g;
    open(FILE,">$cfile.tmp");
    print FILE $filestr;
}

File: Python_Scripting_for_Computational_Science_Third_Edition/py/regex/swap1.py
#!/usr/bin/env python
import re, glob, string
arg = r'[^,]+'
call = r'superLibFunc\s*\(\s*(%s),\s*(%s)\)' % (arg,arg)

cfiles = ['.test1.c']
for cfile in cfiles:
    print 'Treating',cfile
    file = open(cfile, 'r')
    filestr = file.read() # load all lines into a string
    file.close()
    filestr = re.sub(call, r'superLibFunc(\2, \1)', filestr)
    file = open(cfile + '.tmp', 'w')
    file.write(filestr)  # print everything into cfile.tmp
